vortex-duckdb/src/duckdb/vector.rs

```diff
@@ -187,6 +198,26 @@ impl Vector {
         unsafe { cpp::duckdb_free(debug.cast_mut().cast()) };
         Ok(string)
     }
+
+    pub fn list_vector_reserve(&self, required_capacity: u64) -> VortexResult<()> {
+        let state = unsafe { cpp::duckdb_list_vector_reserve(self.as_ptr(), required_capacity) };
+        match state {
+            cpp::duckdb_state::DuckDBSuccess => Ok(()),
+            cpp::duckdb_state::DuckDBError => vortex_bail!("vector was nullptr!"),
+        }
+    }
+
+    pub fn list_vector_get_child(&self) -> Self {
+        unsafe { Self::borrow(cpp::duckdb_list_vector_get_child(self.as_ptr())) }
+    }
+
+    pub fn list_vector_set_size(&self, size: u64) -> VortexResult<()> {
+        let state = unsafe { cpp::duckdb_list_vector_set_size(self.as_ptr(), size) };
+        match state {
+            cpp::duckdb_state::DuckDBSuccess => Ok(()),
+            cpp::duckdb_state::DuckDBError => vortex_bail!("vector was nullptr!"),
```

Comment:
is that the only reason?

Reply:
According to the docs, yes: https://duckdb.org/docs/stable/clients/c/vector.html#duckdb_list_vector_set_size